modules/apps/frontend-js/frontend-js-translation-web/test/TranslationSelector.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {cleanup, render} from '@testing-library/react';
+import React from 'react';
+
+import TranslationSelector from '../src/main/resources/META-INF/resources/TranslationSelector';
+
+const locales = [
+	{
+		label: 'en-US',
+		symbol: 'en-us',
+	},
+	{
+		label: 'es-ES',
+		symbol: 'es-es',
+	},
+	{
+		label: 'fr-FR',
+		symbol: 'fr-fr',
+	},
+	{
+		label: 'hr-HR',
+		symbol: 'hr-hr',
+	},
+];
+
+const spritemap = '/path/to/svg';
+
+const translations = {
+	'en-US': 'Apple',
+	'es-ES': 'Manzana',
+};
+
+describe('TranslationSelector', () => {
+	afterEach(cleanup);
+
+	it('renders', () => {
+		const {container} = render(
+			<TranslationSelector
+				locales={locales}
+				onSelectedLocaleChange={() => {}}
+				selectedLocale={locales[0]}
+				spritemap={spritemap}
+				translations={translations}
+			/>
+		);
+
+		expect(container).toMatchSnapshot();
+	});
+
+	it('renders with a small button', () => {
+		const {container} = render(
+			<TranslationSelector
+				buttonSize="small"
+				locales={locales}
+				onSelectedLocaleChange={() => {}}
+				selectedLocale={locales[0]}
+				spritemap={spritemap}
+				translations={translations}
+			/>
+		);
+
+		expect(container).toMatchSnapshot();
+	});
+
+	it('renders as a language selector', () => {
+		const {container} = render(
+			<TranslationSelector
+				locales={locales}
+				onSelectedLocaleChange={() => {}}
+				selectedLocale={locales[0]}
+				spritemap={spritemap}
+				translations={translations}
+				variant="language selector"
+			/>
+		);
+
+		expect(container).toMatchSnapshot();
+	});
+
+	it('renders as a translation selector', () => {
+		const {container} = render(
+			<TranslationSelector
+				locales={locales}
+				onSelectedLocaleChange={() => {}}
+				selectedLocale={locales[0]}
+				spritemap={spritemap}
+				translations={translations}
+				variant="translation selector"
+			/>
+		);
+
+		expect(container).toMatchSnapshot();
+	});
```

Comment:
You can add a test for testing `onSelectedLocaleChange` callback :)

Reply:
Yes, why not

Reply:
Updated in [`2752f20`](https://github.com/liferay-frontend/liferay-portal/pull/1184/commits/2752f205b152e876ce51c3898d5241a89d802b89)